test/e2e/e2e.go

```diff
@@ -175,6 +191,64 @@ var _ = Describe("validation", func() {
 			})
 		})
 	})
+
+	Context("Creating AddressPool", func() {
```

Comment:
Not related to the tests, but what happens if the user adds two different address pools with different names?

Reply:
that is fine that is what I have in the example and service through annotation can pick whatever pool it wants

Reply:
I am not sure about that, if I understood correctly the reconcile loop will squash the multiple address pools into a single configmap, and only the latest will be rendered.

Reply:
no I added a function in the merge logic that will append the configmap to have multiple pools I tested that as part of #17

Reply:
Ah, thanks for explaining, I missed that part.

So I have another doubt: will this work if you add two addressPools, generate the merged configmap and remove one?
Can you add another test for this scenario? I am afraid the controller reference thing will work only in case of 1:1 mapping

Reply:
I will cover that in UT test cases for the merge not in e2e CI run

Reply:
In unit tests you can't verify the controller reference behaviour because that's a kubernetes thing.
I believe this is a relevant case that needs to be added.

Reply:
I don't need controller I can build mockup data for the current and updated structure to cover all variances , I checking to see how to add table to cover multiple variants so I can get this off my way to focus on the remaining critical issues I have

Reply:
You need that, because the controller reference is supposed to work on 1:1 relationship (and it's currently not working). So deletion needs to be handled manually I believe, probably collecting all the addresspools and re-generating the configmap out of them.

I filed https://github.com/metallb/metallb-operator/issues/32 for that.

Reply:
ok will chase that issue separately